opensearch-operator/controllers/cluster_test.go

```diff
@@ -85,6 +87,12 @@ var _ = Describe("Cluster Reconciler", func() {
 			Expect(k8sClient.Create(context.Background(), &OpensearchCluster)).Should(Succeed())
 		})
 
+		It("should create a ServiceMonitor for the cluster", func() {
+			Eventually(func() error {
+				return k8sClient.Get(context.Background(), client.ObjectKey{Name: OpensearchCluster.Name + "-monitor", Namespace: OpensearchCluster.Namespace}, &prometheus.ServiceMonitor{})
+			}, timeout, interval).Should(Succeed())
```

Comment:
That test is too minimal. Please also check that the created ServiceMonitor is configured with the correct endpoints, scrape interval and check if the secret is created with the correct credentials.